addons/mail/models/mail_link_preview.py

```diff
@@ -1,9 +1,11 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+import re
 import requests
 
 from datetime import timedelta
 from lxml import html
+from werkzeug.urls import url_join
```

Comment:
Deprecated since version 2.3: Will be removed in Werkzeug 2.4. Use urllib.parse.urljoin instead.